### PR TITLE
feat(cts): add e2e for scala

### PR DESCRIPTION
**Pull request by millotp**
## 🧭 What and Why

Add e2e CTS for scala.

### CODE REVIEW

File: templates/scala/tests/requests/requests.mustache [hunk 1]
```diff
@@ -33,6 +40,27 @@ class {{clientPrefix}}Test extends AnyFunSuite {
...
+        region = {{#fallbackToAliasHost}}Some("{{defaultRegion}}"){{/fallbackToAliasHost}}{{^fallbackToAliasHost}}"{{defaultRegion}}"{{/fallbackToAliasHost}}
+        {{/hasRegionalHost}}
+      )
+    } else {
+      val dotenv = Dotenv.configure.directory("../../").load
+      {{client}}(
+        appId = dotenv.get("ALGOLIA_APPLICATION_ID"),
+        apiKey = dotenv.get("ALGOLIA_ADMIN_KEY"){{#hasRegionalHost}},
+        region = {{#fallbackToAliasHost}}Some("{{defaultRegion}}"){{/fallbackToAliasHost}}{{^fallbackToAliasHost}}"{{defaultRegion}}"{{/fallbackToAliasHost}}
+        {{/hasRegionalHost}}
```
Comment (shortcuts) [inline templates/scala/tests/requests/requests.mustache:58]: region can be shared here no?

**Review by shortcuts — COMMENTED** [review]
wonderful

File: templates/scala/tests/requests/requests.mustache (hunk 1 above)
Comment (millotp) [reply templates/scala/tests/requests/requests.mustache]: sure

**Review by shortcuts — APPROVED** [review]
gg!!

**Pull request merged**